canu/utils/inventory.py

```diff
@@ -0,0 +1,86 @@
+# MIT License
+#
+# (C) Copyright [2022] Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+"""Create Nornir Inventory from SLS."""
+import json
+
+import click
+
+from canu.utils.sls import pull_sls_hardware, pull_sls_networks
+
+
+def inventory(username, password, network, sls_json=None):
+    """Build Nornir inventory from sls_input."""
+    inventory = {"groups": "shasta", "hosts": {}}
+    if sls_json:
+        try:
+            input_json = json.load(sls_json)
+        except (json.JSONDecodeError, UnicodeDecodeError):
+            click.secho(
+                f"The file {sls_json.name} is not valid JSON.",
+                fg="red",
+            )
+            return
+        sls_variables = pull_sls_networks(input_json)
+        sls_hardware = pull_sls_hardware(input_json)
+    else:
+        sls_variables = pull_sls_networks()
+        sls_hardware = pull_sls_hardware()
+
+    for k in sls_variables[network + "_IPs"]:
+        if "sw" in k:
+            inventory["hosts"].update(
+                {
+                    k: {
+                        "hostname": str(sls_variables[network + "_IPs"][k]),
+                        "platform": "",
+                        "username": username,
+                        "password": password,
+                    },
+                },
+            )
+    # pull in the platform type from sls hardware data
+    for x in sls_hardware:
```

Comment:
prefer more descriptive variables unless you're trying to do a compact list comprehension or generalized algorithm.  in this case `x` should be something like `node` or `device`.